File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/octoGame/GameLevelsViewModelFactory.java
package com.dreamproject.inwords.presentation.viewScenario.octoGame;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

@Singleton
public class GameLevelsViewModelFactory implements ViewModelProvider.Factory {
    @Inject
    GameLevelsViewModelFactory() {
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(GameLevelsViewModel.class)) {
            //noinspection unchecked
            return (T) new GameLevelsViewModel();
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }

}

File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/octoGame/GameLevelsViewModelFactory.kt
package com.dreamproject.inwords.presentation.viewScenario.octoGame

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.dreamproject.inwords.data.repository.GameRemoteRepository
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GameLevelsViewModelFactory @Inject
internal constructor(private val gameRemoteRepository: GameRemoteRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(GameLevelsViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return GameLevelsViewModel(gameRemoteRepository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}
